Sources/BuildSystemIntegration/CompilationDatabaseBuildSystem.swift

```diff
@@ -118,18 +140,20 @@ package actor CompilationDatabaseBuildSystem: BuiltInBuildSystem {
   package nonisolated var supportsPreparation: Bool { false }
 
   package init?(
-    projectRoot: URL,
-    searchPaths: [RelativePath],
+    configPath: URL,
     connectionToSourceKitLSP: any Connection
-  ) {
-    self.projectRoot = projectRoot
-    self.searchPaths = searchPaths
-    self.connectionToSourceKitLSP = connectionToSourceKitLSP
-    if let compdb = tryLoadCompilationDatabase(directory: projectRoot, additionalSearchPaths: searchPaths) {
+  ) throws {
+    if let compdb = tryLoadCompilationDatabase(file: configPath) {
       self.compdb = compdb
     } else {
       return nil
     }
+    self.connectionToSourceKitLSP = connectionToSourceKitLSP
+
+    self.configPath = configPath
+    self.fileWatchers = [
+      FileSystemWatcher(globPattern: try configPath.filePath, kind: [.change])
```

Comment:
There is one semi-common pattern that we’d be missing now: 

The user has a build that stores `compile_commands.json` in `myfancybuildfolder`. In order to pick it up, they create a symlink from `<project root>/compile_commands.json` to `myfancybuildfolder/compile_commands.json`. Now, we no longer reload the compile commands if `myfancybuildfolder/compile_commands.json` changes.

Though, maybe we should tell these users to use the custom search path config setting (which they can commit since we’ve got `config.json` since 6.0).

Reply:
I slight preference for only watching the config file itself, but if you think that's a common enough pattern then I'm fine switching it back (and I'll add a comment to explain why we're doing that).